components/registry-facade/cmd/run.go

```diff
@@ -109,11 +113,36 @@ var runCmd = &cobra.Command{
 			return docker.NewResolver(resolverOpts)
 		}
 
+		if cfg.ReadinessProbeAddr != "" {
+			// use the first layer as source for the tests
+			if len(cfg.Registry.StaticLayer) < 1 {
+				log.Panic("the configuration file do not contain static layers")
+			}
+
+			staticLayerRef := cfg.Registry.StaticLayer[0].Ref
+			staticLayerHost := strings.Split(staticLayerRef, "/")[0]
+			// Ensure we can resolve DNS queries, can access the reistry and the local app is up and running
+			health := healthcheck.NewHandler()
+			health.AddReadinessCheck("dns", kubernetes.DNSCanResolveProbe(staticLayerHost, 1*time.Second))
+			health.AddReadinessCheck("registry", kubernetes.NetworkIsReachableProbe(fmt.Sprintf("http://%v", staticLayerRef)))
+			health.AddReadinessCheck("registry-facade", kubernetes.NetworkIsReachableProbe(fmt.Sprintf("https://127.0.0.1:%v/%v/base/", cfg.Registry.Port, cfg.Registry.Prefix)))
+
+			health.AddLivenessCheck("dns", kubernetes.DNSCanResolveProbe(staticLayerHost, 1*time.Second))
+			health.AddLivenessCheck("registry", kubernetes.NetworkIsReachableProbe(fmt.Sprintf("http://%v", staticLayerRef)))
```

Comment:
Would it make sense to add a liveness check for `registry-facade`? For example, if we were to add, and it failed, the intent would be to remove the node label to prevent scheduling.

Reply:
edit: updated question to be more clear

Reply:
not necessary, we're just checking that the web server is online for registry-facade, we don't want to pass the traffic through to the underlying registry.